unittest/indent_formatter_test.cpp

```diff
@@ -0,0 +1,103 @@
+#include <sstream>
```

Comment:
This is the other bit of net-new code, and implements unit tests (using stringstreams) to test the manipulator functionality.   Since those manipulators work on any stream, they work on stringstreams for testing and then
I know that they work correctly on filestreams which is how they get used in the actual code.